tests/build_/test_build_request.py

```diff
@@ -2105,3 +2105,70 @@ def test_info_url_format(self):
         assert plugin_value_get(plugins, 'prebuild_plugins',
                                 'add_labels_in_dockerfile', 'args',
                                 'info_url_format') == info_url_format
+
+    @pytest.mark.parametrize('platforms', [None, "x86"])
+    @pytest.mark.parametrize('platform', [None, "x86", "ppc"])
+    @pytest.mark.parametrize('is_auto', [True, False])
+    @pytest.mark.parametrize('scratch', [True, False])
+    def test_check_set_nodeselectors(self, platforms, platform, is_auto, scratch):
+        platform_nodeselectors = {
+            'x86': {
+                'plx86a': 'yes',
+                'plx86b': 'yes'
+            },
+            'ppc': {
+                'plppc1': 'yes',
+                'plppc2': 'yes'
+            },
+            'extra': {
+                'test1': 'yes',
+                'test2': 'yes'
+            }
+        }
+        built_type_nodeselectors = {
+            'auto': {
+                'auto1': 'yes',
+                'auto2': 'yes'
+            },
+            'explicit': {
+                'explicit1': 'yes',
+                'explicit2': 'yes'
+            },
+            'scratch': {
+                'scratch1': 'yes',
+                'scratch2': 'yes'
+            }
+        }
+        nodeselector_expected = {}
+        if not platforms:
+            if platform:
```

Comment:
This is repeating the logic of the app. Instead `nodeselector_expected` should be passed as a param to the function